server.js

```diff
@@ -15,6 +17,131 @@ app.get("/", function (request, response) {
 
 // TODO add your routes and helper functions here
 
-const listener = app.listen(process.env.PORT, function () {
+//get all bookings
+app.get("/bookings", function (request, response) {
+  response.status(200).json(bookings);
+});
+app.get("/bookings/search", function (request, response) {
+  const word = request.query.term;
+  let id_booking = [];
+  const dateEntered = request.query.date;
+  if (word) {
+    id_booking = searchWord(word);
+    if (id_booking.length >= 1) {
+      response.status(200).json(id_booking);
+    } else {
+      response.status(404).json({ message: "no entry found" });
+    }
+  } else if (dateEntered) {
+    id_booking = searchDate(dateEntered);
+    if (id_booking.length >= 1) {
+      response.status(200).json(id_booking);
+    } else {
+      response.status(404).json({ message: "no entry found" });
+    }
+  } else {
+    response.status(404).json({ message: "no entry found" });
+  }
```

Comment:
I'm not sure that 404 is the correct response for either the text search or the date search. If a user is doing get by ID, then they are asking for a specific resources and they expect that the resource exists, so 404 is correct. However, a search may be completely speculative - an empty array is not an error, as the user may not know whether there are any bookings which meet the criteria. There's an explanation of this here: https://moz.com/community/q/topic/20389/no-search-results-found-should-this-return-status-code-404/3